### PR TITLE
[Small] Explicitly exclude .terraform folder in gitignore

### PR DESCRIPTION
**Pull request by lvreynoso**
### Description

The previous patterns to exclude Terraform folders did not filter out the `.terraform/` folder on my machine, so this explicitly excludes it.

#### Issue
[ch<fill_in_issue_number>](https://app.clubhouse.io/genepi/story/<fill_in_issue_number>)

### Test plan

Run `git add .` then `git status` and see that the folder is not included.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -312,6 +312,7 @@ static
 /.terraform.*/
 /.terraform.lock.hcl
 /terraform/variables.tf
+.terraform
```
Comment (jgadling) [inline .gitignore:315]: can/should we also remove the terraform directory from this repo?

Comment (lvreynoso) [reply .gitignore:315]: I'm not sure, are the batch jobs moved over? I'm not sure if anything still relies on that folder

Comment (jgadling) [reply .gitignore:315]: Hm, I think the only terraform code that's still (somewhat) in use from that directory is the s3 buckets. We should import those into our infra repo when we get a chance.

**Review by jgadling — APPROVED** [review]

**Comment by phoenixAja** [comment]
should we get the CI and lint fixed first?

**Pull request merged**